app/models/services/facebook.rb

```diff
@@ -24,14 +24,18 @@ def post_to_facebook(url, body)
 
   def create_post_params(post)
     message = post.message.plain_text_without_markdown
-    if post.photos.any?
-      message += " " + short_post_url(post, protocol: AppConfig.pod_uri.scheme,
-                                            host: AppConfig.pod_uri.authority)
-    end
-
-    {message: message,
-     access_token: access_token,
-     link:  post.message.urls.first
+    post_url = short_post_url(post, protocol: AppConfig.pod_uri.scheme, host: AppConfig.pod_uri.authority)
+
+    message = if message.empty? && post.photos.any?
```

Comment:
When the message is empty, there always must be photos, so you can just remove the `&& post.photos.any?` here, because it's always true if the `message.empty?` is true.

Reply:
Isn't that implied behavior versus explicit behavior?  We only want the post format to be solely the URL if the message is empty and there is a photo.  The way the code reads it states that directly, even if the second check is not necessary.  I'll implement it either way but it seems having more explicit code on intent here is better.

Reply:
> We only want the post format to be solely the URL if the message is empty and there is a photo. 

Well, in fact, I only want to the post format to be solely the URL if the message is empty. Because as soon as there is no message, the "via" format doesn't make sense anymore. The fact that there are photos in the original don't make any difference. At the moment it's only valid with photos when the message is empty, but maybe that changes in the future. What if it's valid to have an empty message when you have a poll for example. Then nobody would change this line here to `message.empty? && post.photos.any? || post.poll.any?` (first because everybody forgot about it, and second because it would make any sense, because the important part for the decision is only `message.empty?`) and it would look very strange if you then have an "empty" post on facebook only containing "(via <post URL>)".

Reply:
Ah true, I had been too laser focused on thinking about it with respect to the photo thing.